test/src/auxkernels/FunctionGradAux.C

```diff
@@ -0,0 +1,45 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "FunctionGradAux.h"
+#include "Function.h"
+#include "MooseMesh.h"
+
+template<>
+InputParameters validParams<FunctionGradAux>()
+{
+  InputParameters params = validParams<AuxKernel>();
+  params.addRequiredParam<FunctionName>("function", "Function used to compute gradient");
+  params.addRequiredParam<unsigned int>("dimension_index", "The dimension x=0|y=1|z=2");
```

Comment:
`MooseEnum` would be better. Would help Atom and peacock user. Conversion into an `unsigned int` would be trivial in the c-tor.